extensions-contrib/influxdb-emitter/src/main/java/org/apache/druid/emitter/influxdb/InfluxdbEmitter.java

```diff
@@ -0,0 +1,231 @@
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+
+
+public class InfluxdbEmitter implements Emitter
+{
+
+  private static final Logger log = new Logger(InfluxdbEmitter.class);
+  private HttpClient influxdbClient;
+  private final InfluxdbEmitterConfig influxdbEmitterConfig;
+  private final AtomicBoolean started = new AtomicBoolean(false);
+  private final ScheduledExecutorService exec = Executors.newScheduledThreadPool(1, new ThreadFactoryBuilder()
+      .setDaemon(true)
+      .setNameFormat("InfluxdbEmitter-%s")
+      .build());
+
+  private final ImmutableSet dimensionWhiteList;
+
+  private final LinkedBlockingQueue<ServiceMetricEvent> eventsQueue;
+
+  public InfluxdbEmitter(InfluxdbEmitterConfig influxdbEmitterConfig)
+  {
+    this.influxdbEmitterConfig = influxdbEmitterConfig;
+    this.influxdbClient = HttpClientBuilder.create().build();
+    this.eventsQueue = new LinkedBlockingQueue<>(influxdbEmitterConfig.getMaxQueueSize());
+
+    this.dimensionWhiteList = ImmutableSet.of(
+        "dataSource",
+        "type",
+        "numMetrics",
+        "numDimensions",
+        "threshold",
+        "dimension",
+        "taskType",
+        "taskStatus",
+        "tier"
+    );
+
+    log.info("constructing influxdb emitter");
+  }
+
+  @Override
+  public void start()
+  {
+    synchronized (started) {
+      if (!started.get()) {
+        exec.scheduleAtFixedRate(
+            new ConsumerRunnable(),
+            influxdbEmitterConfig.getFlushDelay(),
+            influxdbEmitterConfig.getFlushPeriod(),
+            TimeUnit.MILLISECONDS
+        );
+        started.set(true);
+      }
+    }
+  }
+
+  @Override
+  public void emit(Event event)
+  {
+    if (event instanceof ServiceMetricEvent) {
+      ServiceMetricEvent metricEvent = (ServiceMetricEvent) event;
+      try {
+        eventsQueue.put(metricEvent);
+      }
+      catch (InterruptedException exception) {
+        log.error(exception.toString());
+        Thread.currentThread().interrupt();
+      }
+    }
+  }
+
+  public void postToInflux(String payload)
+  {
+    HttpPost post = new HttpPost(
+        "http://" + influxdbEmitterConfig.getHostname()
+        + ":" + influxdbEmitterConfig.getPort()
+        + "/write?db=" + influxdbEmitterConfig.getDatabaseName()
+        + "&u=" + influxdbEmitterConfig.getInfluxdbUserName()
+        + "&p=" + influxdbEmitterConfig.getInfluxdbPassword()
+    );
+
+    post.setEntity(new StringEntity(payload, ContentType.DEFAULT_TEXT));
+    post.setHeader("Content-Type", "application/x-www-form-urlencoded");
+
+    try {
+      influxdbClient.execute(post);
+    }
+    catch (IOException ex) {
+      log.info(ex.toString());
```

Comment:
this swallows the stacktrace.

```suggestion
      log.info(ex, "Failed to post events to InfluxDB.");
```